apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/pac/R4ClaimResponseResourceProviderE2E.java

```diff
@@ -115,6 +117,33 @@ void shouldGetCorrectClaimResponseResourcesByMbiHash() {
     AssertUtils.assertJsonEquals(expected, actual, ignorePatterns);
   }
 
+  /**
+   * Tests to see if the correct response is given when a search is done for {@link ClaimResponse}s
+   * using given mbi and excludeSAMHSA=true, since this does an extra check for samhsa data.
+   */
+  @Test
+  void shouldGetClaimResponseResourcesByMbiHashWithExcludeSamhsaTrue() {
+    IGenericClient fhirClient = ServerTestUtils.get().createFhirClientV2();
+
+    Bundle claimResult =
+        fhirClient
+            .search()
+            .forResource(ClaimResponse.class)
+            .where(
+                Map.of(
+                    "mbi",
+                    Collections.singletonList(new ReferenceParam(RDATestUtils.MBI_OLD_HASH)),
+                    "service-date",
+                    Arrays.asList(new DateParam("gt1970-07-18"), new DateParam("lt1970-07-25")),
+                    "excludeSAMHSA",
+                    Collections.singletonList(new ReferenceParam("true"))))
+            .returnBundle(Bundle.class)
+            .execute();
+
+    // Ensure we got a result and not an exception
+    assertNotNull(claimResult);
```

Comment:
This is a bit rudimentary, and I was attempting to write something better, but the way the test is written uses a json parsing comparitor and apparently some of the expected response does get filtered, so it was going to be a hassle to figure out what the response _should_ be here. For the purposes of this bug I just wanted to make sure we didn't blow up with an E2E test that captures the issue here (and unit tests better test the finer details below)

Reply:
I agree it would be good to expand the test comparison later.  But not today.